scene/2d/position_2d.cpp

```diff
@@ -66,5 +68,31 @@ void Position2D::_notification(int p_what) {
 	}
 }
 
+void Position2D::set_gizmo_extents(float p_extents) {
+	if (p_extents == DEFAULT_GIZMO_EXTENTS) {
+		set_meta("_gizmo_extents_", Variant());
+	} else {
+		set_meta("_gizmo_extents_", p_extents);
+	}
+
+	update();
+}
+
+float Position2D::get_gizmo_extents() const {
+	if (has_meta("_gizmo_extents_")) {
+		return get_meta("_gizmo_extents_");
+	} else {
+		return DEFAULT_GIZMO_EXTENTS;
+	}
+}
+
+void Position2D::_bind_methods() {
+
+	ClassDB::bind_method(D_METHOD("set_gizmo_extents", "extents"), &Position2D::set_gizmo_extents);
+	ClassDB::bind_method(D_METHOD("get_gizmo_extents"), &Position2D::get_gizmo_extents);
+
+	ADD_PROPERTY(PropertyInfo(Variant::REAL, "gizmo_extents", PROPERTY_HINT_NONE, "", PROPERTY_USAGE_EDITOR | PROPERTY_USAGE_INTERNAL), "set_gizmo_extents", "get_gizmo_extents");
```

Comment:
It would be good to add a `PROPERTY_HINT_RANGE` to limit the range to reasonable values (`"0,1000,0.1,or_greater"` maybe?)